### ISSUE TITLE
Can't install operator with glusterfs storage

### ISSUE THREAD

**Issue by alrooney** [CrunchyData/postgres-operator#1539]
** Which example are you working with? **
Installing pgo 4.3.0 and would like to use glusterfs.  I added the following to postgres-operatory.yml:

                  - name: STORAGE10_NAME
                    value: "glusterfs"
                  - name: STORAGE10_ACCESS_MODE
                    value: "ReadWriteMany"
                  - name: STORAGE10_SIZE
                    value: "100Gi"
                  - name: STORAGE10_TYPE
                    value: "dynamic"
                  - name: STORAGE10_CLASS
                    value: "glusterfs-storage"

**What is the current behavior?**
The operator fails to start

```
NAME                                     READY   STATUS             RESTARTS   AGE
pod/pgo-deploy-f288b                     0/1     Error              0          16m
pod/postgres-operator-7696fd75d4-qdsmr   1/4     CrashLoopBackOff   22         16m
```

**What is the expected behavior?**
operator starts correctly with glusterfs

**Other information** (e.g. detailed explanation, related issues,  etc)

**Please tell us about your environment:**

* Operating System:
* Where is this running ( Local , Cloud Provider)
* Storage being used (NFS, Hostpath, Gluster, etc):
* Container Image Tag:
* PostgreSQL Version:
* Platform (Docker, Kubernetes, OpenShift):
* Platform Version:

If possible please run the following on the kubernetes or OpenShift (oc) commands and provide the result:
 kubectl describe yourPodName

```
$ kubectl describe pod/pgo-deploy-f288b
Name:           pgo-deploy-f288b
Namespace:      pgo
Priority:       0
Node:           node-z2499nk-10-0-1-8-prod1-azure/10.0.1.8
Start Time:     Sat, 16 May 2020 21:04:29 -0400
Labels:         controller-uid=5c6cb708-97da-11ea-bb82-000d3a7d2c88
                job-name=pgo-deploy
Annotations:    cni.projectcalico.org/podIP: 10.244.4.16/32
Status:         Failed
IP:             10.244.4.16
IPs:            <none>
Controlled By:  Job/pgo-deploy
Containers:
  pgo-deploy:
    Container ID:  docker://f0cf3caa8bb398892027040fcd83aae9d17b2ec7f7969cd23c43a01e2102dbd4
    Image:         registry.developers.crunchydata.com/crunchydata/pgo-deployer:centos7-4.3.0
    Image ID:      docker-pullable://registry.developers.crunchydata.com/crunchydata/pgo-deployer@sha256:36b81e8dcfc246c4328ea2a0f75136cd1c9bef7e64e142048c92b03d0e088b1e
    Port:          <none>
    Host Port:     <none>
    Command:
      /pgo-deploy.sh
    State:          Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Sat, 16 May 2020 21:04:30 -0400
      Finished:     Sat, 16 May 2020 21:15:02 -0400
    Ready:          False
    Restart Count:  0
    Environment:
      ANSIBLE_CONFIG:                /ansible/ansible.cfg
      ARCHIVE_MODE:                  true
      ARCHIVE_TIMEOUT:               60
      BACKREST:                      true
      BADGER:                        false
      CRUNCHY_DEBUG:                 false
      CREATE_RBAC:                   true
      CCP_IMAGE_PREFIX:              registry.developers.crunchydata.com/crunchydata
      CCP_IMAGE_TAG:                 centos7-12.2-4.3.0
      DB_PASSWORD_LENGTH:            24
      DB_PORT:                       5432
      DB_REPLICAS:                   0
      DB_USER:                       testuser
      DEFAULT_INSTANCE_MEMORY:       128Mi
      DEFAULT_PGBACKREST_MEMORY:     
      DEFAULT_PGBOUNCER_MEMORY:      
      DEPLOY_ACTION:                 install
      DISABLE_AUTO_FAILOVER:         false
      EXPORTERPORT:                  9187
      HOME:                          /tmp
      METRICS:                       false
      NAMESPACE:                     pgo
      NAMESPACE_MODE:                dynamic
      PGBADGERPORT:                  10000
      PGO_ADMIN_PASSWORD:            password
      PGO_ADMIN_PERMS:               *
      PGO_ADMIN_ROLE_NAME:           pgoadmin
      PGO_ADMIN_USERNAME:            admin
      PGO_CLIENT_VERSION:            v4.3.0
      PGO_IMAGE_PREFIX:              registry.developers.crunchydata.com/crunchydata
      PGO_IMAGE_TAG:                 centos7-4.3.0
      PGO_INSTALLATION_NAME:         devtest
      PGO_OPERATOR_NAMESPACE:        pgo
      SCHEDULER_TIMEOUT:             3600
      BACKREST_STORAGE:              glusterfs
      BACKUP_STORAGE:                glusterfs
      PRIMARY_STORAGE:               glusterfs
      REPLICA_STORAGE:               glusterfs
      WAL_STORAGE:                   
      STORAGE1_NAME:                 hostpathstorage
      STORAGE1_ACCESS_MODE:          ReadWriteMany
      STORAGE1_SIZE:                 1G
      STORAGE1_TYPE:                 create
      STORAGE2_NAME:                 replicastorage
      STORAGE2_ACCESS_MODE:          ReadWriteMany
      STORAGE2_SIZE:                 700M
      STORAGE2_TYPE:                 create
      STORAGE3_NAME:                 nfsstorage
      STORAGE3_ACCESS_MODE:          ReadWriteMany
      STORAGE3_SIZE:                 1G
      STORAGE3_TYPE:                 create
      STORAGE3_SUPPLEMENTAL_GROUPS:  65534
      STORAGE4_NAME:                 nfsstoragered
      STORAGE4_ACCESS_MODE:          ReadWriteMany
      STORAGE4_MATCH_LABEL:          crunchyzone=red
      STORAGE4_TYPE:                 create
      STORAGE4_SUPPLEMENTAL_GROUPS:  65534
      STORAGE5_NAME:                 storageos
      STORAGE5_ACCESS_MODE:          ReadWriteOnce
      STORAGE5_SIZE:                 5Gi
      STORAGE5_TYPE:                 dynamic
      STORAGE5_CLASS:                fast
      STORAGE6_NAME:                 primarysite
      STORAGE6_ACCESS_MODE:          ReadWriteOnce
      STORAGE6_SIZE:                 4G
      STORAGE6_TYPE:                 dynamic
      STORAGE6_CLASS:                primarysite
      STORAGE7_NAME:                 alternatesite
      STORAGE7_ACCESS_MODE:          ReadWriteOnce
      STORAGE7_SIZE:                 4G
      STORAGE7_TYPE:                 dynamic
      STORAGE7_CLASS:                alternatesite
      STORAGE8_NAME:                 gce
      STORAGE8_ACCESS:               ReadWriteOnce
      STORAGE8_SIZE:                 300M
      STORAGE8_TYPE:                 dynamic
      STORAGE8_CLASS:                standard
      STORAGE9_NAME:                 rook
      STORAGE9_ACCESS_MODE:          ReadWriteOnce
      STORAGE9_SIZE:                 1Gi
      STORAGE9_TYPE:                 dynamic
      STORAGE9_CLASS:                rook-ceph-block
      STORAGE10_NAME:                glusterfs
      STORAGE10_ACCESS_MODE:         ReadWriteMany
      STORAGE10_SIZE:                100Gi
      STORAGE10_TYPE:                dynamic
      STORAGE10_CLASS:               glusterfs-storage
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from pgo-deployer-sa-token-s59lh (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  pgo-deployer-sa-token-s59lh:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  pgo-deployer-sa-token-s59lh
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type    Reason     Age   From                                        Message
  ----    ------     ----  ----                                        -------
  Normal  Scheduled  22m   default-scheduler                           Successfully assigned pgo/pgo-deploy-f288b to node-z2499nk-10-0-1-8-prod1-azure
  Normal  Pulled     22m   kubelet, node-z2499nk-10-0-1-8-prod1-azure  Container image "registry.developers.crunchydata.com/crunchydata/pgo-deployer:centos7-4.3.0" already present on machine
  Normal  Created    22m   kubelet, node-z2499nk-10-0-1-8-prod1-azure  Created container pgo-deploy
  Normal  Started    22m   kubelet, node-z2499nk-10-0-1-8-prod1-azure  Started container pgo-deploy
```

 kubectl describe pvc
- no pvcs
 kubectl get nodes
```
$ kubectl get nodes
NAME                                STATUS   ROLES    AGE     VERSION
node-z2499nk-10-0-1-4-prod1-azure   Ready    master   3d13h   v1.14.1
node-z2499nk-10-0-1-5-prod1-azure   Ready    master   3d13h   v1.14.1
node-z2499nk-10-0-1-6-prod1-azure   Ready    master   3d13h   v1.14.1
node-z2499nk-10-0-1-7-prod1-azure   Ready    <none>   3d13h   v1.14.1
node-z2499nk-10-0-1-8-prod1-azure   Ready    <none>   3d12h   v1.14.1
```
 kubectl log yourPodName
```
$ kubectl logs pod/pgo-deploy-f288b

PLAY [Deploy Crunchy PostgreSQL Operator] **************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [pgo-preflight : fail] ****************************************************
skipping: [localhost]

TASK [pgo-preflight : Check Operating System] **********************************
changed: [localhost]

TASK [pgo-preflight : assert] **************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [pgo-preflight : assert] **************************************************
skipping: [localhost]

TASK [pgo-preflight : assert] **************************************************
skipping: [localhost]

TASK [pgo-preflight : assert] **************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [pgo-preflight : include_tasks] *******************************************
skipping: [localhost]

TASK [pgo-preflight : include_tasks] *******************************************
included: /ansible/roles/pgo-preflight/tasks/check_kubernetes.yml for localhost

TASK [pgo-preflight : Check if the kubectl command is installed] ***************
changed: [localhost]

TASK [pgo-preflight : Ensure kubectl is installed] *****************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [pgo-preflight : include_tasks] *******************************************
included: /ansible/roles/pgo-preflight/tasks/check_inventory.yml for localhost

TASK [pgo-preflight : Check if mandatory inventory file variables are defined] ***
skipping: [localhost] => (item=pgo_operator_namespace) 
skipping: [localhost] => (item=pgo_installation_name) 
skipping: [localhost] => (item=pgo_admin_username) 
skipping: [localhost] => (item=pgo_admin_password) 
skipping: [localhost] => (item=pgo_admin_role_name) 
skipping: [localhost] => (item=pgo_admin_perms) 
skipping: [localhost] => (item=ccp_image_prefix) 
skipping: [localhost] => (item=ccp_image_tag) 
skipping: [localhost] => (item=pgo_image_prefix) 
skipping: [localhost] => (item=pgo_image_tag) 
skipping: [localhost] => (item=disable_auto_failover) 
skipping: [localhost] => (item=backrest) 
skipping: [localhost] => (item=badger) 
skipping: [localhost] => (item=metrics) 
skipping: [localhost] => (item=archive_mode) 
skipping: [localhost] => (item=archive_timeout) 
skipping: [localhost] => (item=db_password_length) 
skipping: [localhost] => (item=create_rbac) 
skipping: [localhost] => (item=db_port) 
skipping: [localhost] => (item=db_replicas) 
skipping: [localhost] => (item=db_user) 
skipping: [localhost] => (item=backrest_storage) 
skipping: [localhost] => (item=backup_storage) 
skipping: [localhost] => (item=primary_storage) 
skipping: [localhost] => (item=replica_storage) 
skipping: [localhost] => (item=pgo_client_version) 
skipping: [localhost] => (item=pgbadgerport) 
skipping: [localhost] => (item=exporterport) 
skipping: [localhost] => (item=scheduler_timeout) 
skipping: [localhost] => (item=namespace_mode) 

TASK [pgo-operator : Set output directory fact] ********************************
ok: [localhost]

TASK [pgo-operator : Ensure output directory exists] ***************************
changed: [localhost]

TASK [pgo-operator : include_tasks] ********************************************
skipping: [localhost] => (item=openshift_auth.yml) 
skipping: [localhost] => (item=openshift.yml) 

TASK [pgo-operator : include_tasks] ********************************************
included: /ansible/roles/pgo-operator/tasks/kubernetes_auth.yml for localhost
included: /ansible/roles/pgo-operator/tasks/kubernetes.yml for localhost

TASK [pgo-operator : Set the Kubernetes Context] *******************************
skipping: [localhost]

TASK [pgo-operator : Get Namespace Details] ************************************
changed: [localhost]

TASK [pgo-operator : Create PGO Namespace] *************************************
skipping: [localhost]

TASK [pgo-operator : include_tasks] ********************************************
included: /ansible/roles/pgo-operator/tasks/certs.yml for localhost

TASK [pgo-operator : Ensure directory exists for local self-signed TLS certs.] ***
ok: [localhost]

TASK [pgo-operator : Generate RSA Key] *****************************************
changed: [localhost]

TASK [pgo-operator : Generate CSR] *********************************************
changed: [localhost]

TASK [pgo-operator : Generate Self-signed Certificate] *************************
changed: [localhost]

TASK [pgo-operator : Ensure /tmp/.pgo/pgo Directory Exists] ********************
ok: [localhost]

TASK [pgo-operator : Copy certificates to /tmp/.pgo/pgo] ***********************
changed: [localhost]

TASK [pgo-operator : Copy keys to /tmp/.pgo/pgo] *******************************
changed: [localhost]

TASK [pgo-operator : Use kubectl or oc] ****************************************
ok: [localhost]

TASK [pgo-operator : include_tasks] ********************************************
included: /ansible/roles/pgo-operator/tasks/namespace.yml for localhost

TASK [pgo-operator : Create Watched Namespaces] ********************************
changed: [localhost] => (item=pgo)

TASK [pgo-operator : include_tasks] ********************************************
included: /ansible/roles/pgo-operator/tasks/crds.yml for localhost

TASK [pgo-operator : Check if PGCluster CRD Is Installed] **********************
changed: [localhost]

TASK [pgo-operator : Create PGClusters CRD] ************************************
skipping: [localhost]

TASK [pgo-operator : Check if PGPolicies CRD Is Installed] *********************
changed: [localhost]

TASK [pgo-operator : Create PGPolicies CRD] ************************************
skipping: [localhost]

TASK [pgo-operator : Check if PGReplicas CRD Is Installed] *********************
changed: [localhost]

TASK [pgo-operator : Create PGReplicas CRD] ************************************
skipping: [localhost]

TASK [pgo-operator : Check if PGTasks CRD Is Installed] ************************
changed: [localhost]

TASK [pgo-operator : Create PGTasks CRD] ***************************************
skipping: [localhost]

TASK [pgo-operator : Template PGO Admin Credentials] ***************************
changed: [localhost]

TASK [pgo-operator : Create PGO Admin Credentials] *****************************
changed: [localhost]

TASK [pgo-operator : Template PGO Admin Role & Permissions] ********************
changed: [localhost]

TASK [pgo-operator : Create PGO Admin Role & Permissions] **********************
changed: [localhost]

TASK [pgo-operator : Template PGO Service Account] *****************************
changed: [localhost]

TASK [pgo-operator : Create PGO Service Account] *******************************
changed: [localhost]

TASK [pgo-operator : Template Cluster RBAC (namespace_mode 'dynamic')] *********
changed: [localhost]

TASK [pgo-operator : Create Cluster RBAC (namespace_mode 'dynamic')] ***********
changed: [localhost]

TASK [pgo-operator : Template Cluster RBAC (namespace_mode 'readonly')] ********
skipping: [localhost]

TASK [pgo-operator : Create Cluster RBAC (namespace_mode 'readonly')] **********
skipping: [localhost]

TASK [pgo-operator : Cluster Roles Disabled (namespace_mode 'disabled')] *******
skipping: [localhost]

TASK [pgo-operator : Create PGO Image Pull Secret] *****************************
skipping: [localhost]

TASK [pgo-operator : Create cluster-admin Cluster Role Binding for PGO Service Account] ***
skipping: [localhost]

TASK [pgo-operator : Template PGO RBAC] ****************************************
changed: [localhost]

TASK [pgo-operator : Create PGO RBAC] ******************************************
changed: [localhost]

TASK [pgo-operator : Template Local PGO User] **********************************
changed: [localhost]

TASK [pgo-operator : Template PGO Configuration] *******************************
changed: [localhost]

TASK [pgo-operator : Template PGO Service Configuration] ***********************
changed: [localhost]

TASK [pgo-operator : Create PGO BackRest Repo Secret] **************************
changed: [localhost]

TASK [pgo-operator : Create PGO API Secret] ************************************
changed: [localhost]

TASK [pgo-operator : Create PGO ConfigMap] *************************************
changed: [localhost]

TASK [pgo-operator : Create PGO Service] ***************************************
changed: [localhost]

TASK [pgo-operator : Template PGO Deployment] **********************************
changed: [localhost]

TASK [pgo-operator : Deploy PGO] ***********************************************
changed: [localhost]

TASK [pgo-operator : Download PGO Linux Client] ********************************
skipping: [localhost]

TASK [pgo-operator : Download PGO MacOS Client] ********************************
skipping: [localhost]

TASK [pgo-operator : Wait for PGO to finish deploying] *************************
fatal: [localhost]: FAILED! => {"ansible_job_id": "104251771367.1366", "changed": true, "cmd": ["kubectl", "rollout", "status", "deployment/postgres-operator", "-n", "pgo"], "delta": "0:10:00.270268", "end": "2020-05-17 01:14:58.018322", "finished": 1, "msg": "non-zero return code", "rc": 1, "start": "2020-05-17 01:04:57.748054", "stderr": "error: deployment \"postgres-operator\" exceeded its progress deadline", "stderr_lines": ["error: deployment \"postgres-operator\" exceeded its progress deadline"], "stdout": "Waiting for deployment \"postgres-operator\" rollout to finish: 0 of 1 updated replicas are available...", "stdout_lines": ["Waiting for deployment \"postgres-operator\" rollout to finish: 0 of 1 updated replicas are available..."]}

PLAY RECAP *********************************************************************
localhost                  : ok=48   changed=33   unreachable=0    failed=1    skipped=19   rescued=0    ignored=0 
```

**Comment by alrooney** [comment]
Ok.  Think operator is working now... Looks like I needed to use one of the existing STORAGE definitions instead of creating a new one i.e. I changed STORAGE9 to use glusterfs settings.  Anyway, I am past error above... Now to actually create a db...

**Comment by jkatz** [comment]
This is a known issue that is fixed in 2ebbaf2 and will be conveniently available in 4.3.1

Thanks!
